probe-rs/src/flash/flasher.rs

```diff
@@ -182,7 +182,23 @@ impl<'a> Flasher<'a> {
         let mut data = vec![0; algo.instructions.len()];
         flasher.probe.read_block32(algo.load_address, &mut data)?;
 
-        assert_eq!(&algo.instructions, &data.as_slice());
+        for (offset, (original, read_back)) in algo.instructions.iter().zip(data.iter()).enumerate()
+        {
+            if original != read_back {
+                eprintln!(
+                    "Failed to verify flash algorithm. Data mismatch at address {:#08x}",
+                    algo.load_address + (4 * offset) as u32
+                );
+                eprintln!("Original instruction: {:#08x}", original);
```

Comment:
This should be converted into a log statement instead of an stderr. Or is this intended because of the panic (Which I am not sure is ok).

Reply:
See #100